src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -77,6 +68,212 @@ namespace QgsWms
+    httpElement.appendChild( getElement );
+    QDomElement olResourceElement = doc.createElement( QStringLiteral( "OnlineResource" )/*wms:OnlineResource*/ );
+    olResourceElement.setAttribute( QStringLiteral( "xmlns:xlink" ), QStringLiteral( "http://www.w3.org/1999/xlink" ) );
+    olResourceElement.setAttribute( QStringLiteral( "xlink:type" ), QStringLiteral( "simple" ) );
+    olResourceElement.setAttribute( QStringLiteral( "xlink:href" ), hrefString );
+    getElement.appendChild( olResourceElement );
+
+    //wms:GetMap
+    elem = doc.createElement( QStringLiteral( "GetMap" )/*wms:GetMap*/ );
+    appendFormat( elem, QStringLiteral( "image/jpeg" ) );
+    appendFormat( elem, QStringLiteral( "image/png" ) );
+    appendFormat( elem, QStringLiteral( "image/png; mode=16bit" ) );
+    appendFormat( elem, QStringLiteral( "image/png; mode=8bit" ) );
+    appendFormat( elem, QStringLiteral( "image/png; mode=1bit" ) );
+    appendFormat( elem, QStringLiteral( "application/dxf" ) );
+    elem.appendChild( dcpTypeElement.cloneNode().toElement() ); //this is the same as for 'GetCapabilities'
+    requestElement.appendChild( elem );
+
+    //wms:GetFeatureInfo
+    elem = doc.createElement( QStringLiteral( "GetFeatureInfo" ) );
+    appendFormat( elem, QStringLiteral( "text/plain" ) );
+    appendFormat( elem, QStringLiteral( "text/html" ) );
+    appendFormat( elem, QStringLiteral( "text/xml" ) );
+    appendFormat( elem, QStringLiteral( "application/vnd.ogc.gml" ) );
+    appendFormat( elem, QStringLiteral( "application/vnd.ogc.gml/3.1.1" ) );
+    elem.appendChild( dcpTypeElement.cloneNode().toElement() ); //this is the same as for 'GetCapabilities'
+    requestElement.appendChild( elem );
+
+    //wms:GetLegendGraphic
+    elem = doc.createElement(( version == QLatin1String( "1.1.1" ) ? "GetLegendGraphic" : "sld:GetLegendGraphic" )/*wms:GetLegendGraphic*/ );
+    appendFormat( elem, QStringLiteral( "image/jpeg" ) );
+    appendFormat( elem, QStringLiteral( "image/png" ) );
+    elem.appendChild( dcpTypeElement.cloneNode().toElement() ); // this is the same as for 'GetCapabilities'
+    requestElement.appendChild( elem );
+
+    //wms:DescribeLayer
+    elem = doc.createElement(( version == QLatin1String( "1.1.1" ) ? "DescribeLayer" : "sld:DescribeLayer" )/*wms:GetLegendGraphic*/ );
+    appendFormat( elem, QStringLiteral( "text/xml" ) );
+    elem.appendChild( dcpTypeElement.cloneNode().toElement() ); // this is the same as for 'GetCapabilities'
+    requestElement.appendChild( elem );
+
+    //wms:GetStyles
+    elem = doc.createElement(( version == QLatin1String( "1.1.1" ) ? "GetStyles" : "qgs:GetStyles" )/*wms:GetStyles*/ );
+    appendFormat( elem, QStringLiteral( "text/xml" ) );
+    elem.appendChild( dcpTypeElement.cloneNode().toElement() ); //this is the same as for 'GetCapabilities'
+    requestElement.appendChild( elem );
+
+    if ( projectSettings ) //remove composer templates from GetCapabilities in the long term
```

Comment:
I don't know: these are original comments. Functionaly nothing has changed, I have only restructured parts of the original code.